test_rclcpp/test/test_allocator.cpp

```diff
@@ -255,18 +255,18 @@ struct InstrumentedAllocator : public allocator_pointer_traits<T>
 
   // construct and destroy should not be required to implement allocator_traits,
   // this is a bug in gcc 4.8 for usage in std::pair, std::map
-  template<typename _Up, typename ... _Args>
-  void
-  construct(_Up * __p, _Args && ... __args)
+  template<typename U, typename ... Args>
+  typename std::enable_if<!std::is_const<U>::value>::type
```

Comment:
Just a style comment, you can add the `enable_if` as a template argument so that the function signature stays the same, e.g.:

```c++
template<typename _Up, typename ... _Args, typename std::enable_if<!std::is_const<U>::value>::type * = nullptr>
void
construct(_Up * __p, _Args && ... __args)
```